### PR TITLE
Add Merge Utility

### PR DESCRIPTION
**Pull request by roshankern**
This PR is ready for review!

In this PR, a merge utility is added to `IDR_stream` that merges one batch of CP and DP data or entire CP and DP runs.

### CODE REVIEW

File: README.md
Comment (d33bs) [inline README.md]: Seeing these appear within the documentation made me wonder if Sphinx [`autodoc`](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) might be useful. I recognize this might be outside the scope of this PR, and maybe a simple way to add some of this functionality would be to link to the related file and line number where the function is defined (selecting the line number in Github will provide you with an anchored link).

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py:193]: Noticing how `merge_CP_DP_run_data` differs from `merge_CP_DP_batch_data` with returns, should this function provide a return of some kind? If not, consider renaming to differentiate from `merge_CP_DP_batch_data` to indicate that the data are saved in addition to being merged.

File: example_notebooks/example_scripts/example_dp.py
Comment (d33bs) [inline example_notebooks/example_scripts/example_dp.py]: Depending on how compatible this may be, consider using a tilde (`~`) here to reference the home directory on any system instead of using the absolute path for specific users. (also applies to other areas in this file)
```suggestion
aspera_path = pathlib.Path("~/.aspera/ascli/sdk/ascp")
```

File: example_notebooks/example_scripts/example_cp.py
Comment (d33bs) [inline example_notebooks/example_scripts/example_cp.py]: Noticing how different these paths may sometimes be, would it make sense to outline expectations on where these resources should be stored in the context of this project (if this hasn't already been done)?

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py]:
```suggestion
        # make location for dp match the closest cp location (distance minimized with hypotenuse)
```

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py:81]: Would there ever be metadata columns which have different names between `cp_columns` and `dp_columns` but should still be excluded from results?

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py:99]: Is there ever a chance that the `Metadata_DNA` column data would not match? If so, consider a quick check for raising an exception on those occasions.

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py:93]: Consider using [`pd.DataFrame.add_prefix()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.add_prefix.html) here for this work.

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py]: Consider casting these data fields earlier instead of on-the-fly here, grouping data changes together with others (such as column selection and renaming). The same might apply for the column itself (preparing all locations and image paths in advance of the loop). Doing this may help with readability or performance.

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py]: Consider moving these nested lambda's to a non-anonymous function (for ex. using `def full_loc_map()`, etc) to increase readability.

Comment (d33bs) [inline idrstream/merge_CP_DP.py]: Could this be performed in advance of entering the loop to decrease the number of times this work is performed? It may be a balance of performance considerations, and if so, please defer to what's best.

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py:135]: Could the `drop`'s be grouped together to save a line or two (perhaps taking advantage of `list` concatenation)?

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py]: Seeing the work to remove warnings or raises, consider providing details about why this is necessary. If this isn't necessary for the work and you could use other methods instead, I recommend doing this to avoid the custom controls of the warnings.

File: idrstream/merge_CP_DP.py
Comment (d33bs) [inline idrstream/merge_CP_DP.py:105]: Could these merges be performed without this loop, making references where appropriate to each dataset's unique `image_path` values?

**Review by d33bs — COMMENTED** [review]
Nice job! I left a few comments and suggestions and would seek your input about whether and how to address these. Please don't hesitate to let me know if you have any questions.

File: README.md
Comment (roshankern) [reply README.md]: I agree that the Sphinx `autodoc` is a bit outside the scope of this PR. Link to related file with line number has been included in https://github.com/WayScience/IDR_stream/pull/15/commits/80bdf8004e7d5af2f423643ea727b9ee79601cbc

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py:193]: Done in https://github.com/WayScience/IDR_stream/pull/15/commits/3188b78ac643d2cfeb1a375bfe67d48d62885e93

File: example_notebooks/example_scripts/example_dp.py
Comment (roshankern) [reply example_notebooks/example_scripts/example_dp.py]: Done in https://github.com/WayScience/IDR_stream/pull/15/commits/098c83c22189ecbd5ac1f38b231a33fc499e9ff4

File: example_notebooks/example_scripts/example_cp.py
Comment (roshankern) [reply example_notebooks/example_scripts/example_cp.py]: Done in https://github.com/WayScience/IDR_stream/pull/15/commits/098c83c22189ecbd5ac1f38b231a33fc499e9ff4. 
This was already outlined in the readme but executed poorly in this notebook.

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py:81]: Possibly... but I think trying to include those here goes a bit out of the scope of this PR.

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py:93]: I don't think `add_prefix` would work here as we only want to add the prefix to columns that either have CP data or DP data.

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py]: I think in this case it is best to perform this within the loop in case some images have cells that are very close.

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py:135]: I don't think so...
We don't want to remove `Full_Location` before merge as this is the column that is merged on.
Leaving the metadata to remove after the merge results in every metadata column being named with a prefix to distinguish the duplicate columns.

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py:105]: Although it would be possible to complete this merge without the loop, I think the for loop makes the code very readable and understandable and is worth keeping for the merge function.

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py]: Done in https://github.com/WayScience/IDR_stream/pull/15/commits/e6a24c15f9d0abe772e43469f902a40d7431a0c2

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py]: Done in https://github.com/WayScience/IDR_stream/pull/15/commits/e6a24c15f9d0abe772e43469f902a40d7431a0c2

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py:99]: Yep... done in https://github.com/WayScience/IDR_stream/pull/15/commits/e6a24c15f9d0abe772e43469f902a40d7431a0c2

File: idrstream/merge_CP_DP.py
Comment (d33bs) [reply idrstream/merge_CP_DP.py:81]: Thanks - it could be that another approach avoiding self-inference could improve things, for example if all the columns include a specific string or follow another similar pattern. I'd defer to you on what's best with the given datasets involved.

**Review by d33bs — APPROVED** [review]

File: idrstream/merge_CP_DP.py
Comment (roshankern) [reply idrstream/merge_CP_DP.py]: Done in https://github.com/WayScience/IDR_stream/pull/15/commits/5036262fe8fa576047173e7caa168d33f7ce67c7

**Pull request merged**
